data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java

```diff
@@ -29,6 +31,9 @@
     private X509Certificate x509Certificate;
     private PrivateKey privateKey;
     private AuthenticationType authenticationType;
+    private AuthenticationResult lastAuthenticationResult;
+    ExecutorService service = Executors.newSingleThreadExecutor();
+    AuthenticationContext context;
```

Comment:
Did you test it in multiple threads environment?
I am not sure what will happen if multiple threads will share this service.